### ISSUE TITLE
There is no way to delete a tool that was added to a thread using "Add Tools"

### ISSUE THREAD

**Issue by sangee2004** [gptscript-ai/desktop#78]
We have the ability to "Add Tools" dynamically when we are chatting with an assistant.

There is no option available to delete the tools that was added using "Add Tools" if the user wants to remove this tool from this thread.

We should provide this ability to delete the tools that was added using "Add Tools"

**Comment by thedadams** [comment]
You can remove tools: click the green checkmark and the tool will be removed.

![Image](https://github.com/user-attachments/assets/176fc9d6-9c07-475a-81be-e3ad3ed116da)



![Image](https://github.com/user-attachments/assets/a6ff8226-d4f4-4377-8eb9-b9a0025bd166)

**Comment by sangee2004** [comment]
I have seen this work in previous build but with latest build this seems to be broken and tracked in https://github.com/gptscript-ai/desktop/issues/221.

Will close this issue once i am able to validate this workflow.

**Comment by sangee2004** [comment]
Tested with latest build with commit id - `731700663e`

I am able to click on the tools that have been added to get it removed. But i see the tool calls continue to work even after tool is removed.

**Steps to reproduce the problem:**
1. Start a chat with `Tildy` and using `Add tools` and  add `outllook mail` tool.
2. Ask for information about the mail and we see it work.
3. Using "Add Tool" option , click on the selected outllook mail` tool and wait for it to be unselected and close the tool catalog.
4. We get a confirmation message about tool being removed in the chat window.
5. Ask for information about the mail and we see it continue to work.

![Image](https://github.com/user-attachments/assets/3b05f071-c38b-4115-b283-a44bb4b6f511)

**Expected Behavior**
This should not have succeeded since we dont have access to the mail tool anymore.

6. Quit the thread and relaunch the thread.

7. Ask for information about the mail and we see it fail with this error:

![Image](https://github.com/user-attachments/assets/23f708e9-84ab-4779-9228-db34ea6259a1)

Is all this  because of LLM halucinating this call ?

**Comment by thedadams** [comment]
PR: https://github.com/gptscript-ai/desktop/pull/245

**Comment by sangee2004** [comment]
Tested with electron build - `b4977d0`

I am able to remove the tools from chat thread by unselecting the tool from `tool catalog` page.

Even after the tool gets removed , LLM keeps calling the tool when there is some queries relating to the deleted tool's expertise. In my case I added outlook mail tool and then removed it.

If i ask any question relating to mail, I see a call being made to email tool call and it fails to fetch any information as expected.

<img width="714" alt="Screenshot 2024-08-23 at 1 21 25 PM" src="https://github.com/user-attachments/assets/a2b01689-caa1-49d2-bbc5-160b1d3ae42f">

Why would LLM be making the tool call to email if this was not even available to it at this time ?
Can we prevent the call from being made to the deleted tools ?
